crates/sound/src/lib.rs

```diff
@@ -1,73 +1,196 @@
+// Manos Pitsidianakis <manos.pitsidianakis@linaro.org>
 // SPDX-License-Identifier: Apache-2.0 or BSD-3-Clause
 
-mod audio_backends;
-mod vhu_sound;
-mod virtio_sound;
+pub mod audio_backends;
+pub mod device;
+pub mod stream;
+pub mod virtio_sound;
 
-use std::io::{Error as IoError, ErrorKind};
-use std::sync::Arc;
+use std::{
+    convert::TryFrom,
+    io::{Error as IoError, ErrorKind},
+    sync::Arc,
+};
 
 use log::{info, warn};
+pub use stream::Stream;
 use thiserror::Error as ThisError;
 use vhost::{vhost_user, vhost_user::Listener};
-use vhost_user_backend::VhostUserDaemon;
-use vm_memory::{GuestMemoryAtomic, GuestMemoryMmap, Le32, VolatileSlice};
+use vhost_user_backend::{VhostUserDaemon, VringRwLock, VringT};
+use virtio_sound::*;
+use vm_memory::{
+    ByteValued, Bytes, GuestMemoryAtomic, GuestMemoryLoadGuard, GuestMemoryMmap, Le32,
+};
 
-use crate::vhu_sound::VhostUserSoundBackend;
+use crate::device::VhostUserSoundBackend;
 
+pub const SUPPORTED_FORMATS: u64 = 1 << VIRTIO_SND_PCM_FMT_U8
```

Comment:
I wonder if those shouldn't be actually enums

Reply:
They are actually not, they are `u8`.